server/src/views/frame.js

```diff
@@ -222,6 +222,7 @@ class Frame extends React.Component {
 
   clickFullPageButton(e) {
     e.preventDefault();
+    history.pushState({state: "frame"}, "", "#frame");
```

Comment:
I did forget about leaving room for the button when you click it; it does make sense for it not to display for a page viewer.  I don't have a very good idea for that.  Except for some hacky flag or something.